src/reducers/transactionReducer.ts

```diff
@@ -160,16 +160,27 @@ const setReceiver = (state: TransactionState, payload: ReceiverPayload): Transac
 
 export default function transactionReducer(state: TransactionState, action: TransactionsActionType): TransactionState {
   switch (action.type) {
-    case TransactionActionTypes.SET_ESTIMATED_FEE:
+    case TransactionActionTypes.SET_ESTIMATED_FEE: {
+      const { estimatedFeeError, estimatedFeeLoading, srcChainDecimals } = action.payload;
+
+      let estimatedFee = null;
+      let estimatedFeeTransformed = null;
+
+      if (!estimatedFeeError && !estimatedFeeLoading && action.payload.estimatedFee) {
+        estimatedFee = action.payload.estimatedFee;
+        estimatedFeeTransformed = transformToBaseUnit(estimatedFee || '0', srcChainDecimals);
+      }
+
       return {
         ...state,
-        estimatedFee: action.payload.estimatedFeeError ? null : action.payload.estimatedFee,
-        estimatedFeeError: action.payload.estimatedFeeError,
-        estimatedFeeLoading: action.payload.estimatedFeeLoading,
-        transactionReadyToExecute: action.payload.estimatedFeeLoading
-          ? false
-          : isReadyToExecute(state) && action.payload.estimatedFee
+        estimatedFee,
+        estimatedFeeTransformed,
```

Comment:
I do not think there is a need to carry across the dispatcher the estimatedFee transformed. As soon as there is the estimatedFee, it can be transformed wherever needed (and that is I think only in 1 place)

Reply:
Yeah, the `transformed` stuff seems purely like a display thing. Also "transformed" is super ambiguous it's not really clear what exactly is happening - seeing a variable like this in the data model seems like something that serves a purpose for exactly one specific component, hence it most likely does not really fit there.

Reply:
Agree guys, it's too overwhelming for the reducer to keep this display value.